From 3f8ee92775467fd676b36ec18aa6a590fad404f6 Mon Sep 17 00:00:00 2001
From: "psathiyendran@maxlinear.com" <psathiyendran@maxlinear.com>
Date: Tue, 1 Jul 2025 18:07:42 +0800
Subject: [PATCH] WLANRTSYS-90145 [WiFi-6E Certification][UPDK][HE-4.3.3.1_6G]:
 AP didn't send re-association frame to STA after injecting a correctly
 protected disassociation frame from STA to the AP

Issue : AP sends protected disassociation frame to station, but still
station sends the data packet, for that AP has to send the "CLASS 3
Non Associated Sta" deauth frame, which is not happening.

Root-Cause : AP sends only disassociation frame for Non Associated
station, it was sending disassociation and deauth before

Regression : WLANRTSYS-57048

Fix : If non-associated frame sends packet, made both dissassoc & deauth
to be sent as before.
---
 src/ap/ieee802_11.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 56fd15efb..652fccd26 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -7822,7 +7822,6 @@ void ieee802_11_rx_from_unknown(struct hostapd_data *hapd, const u8 *src,
 		hostapd_drv_sta_disassoc(
 			hapd, src,
 			WLAN_REASON_CLASS3_FRAME_FROM_NONASSOC_STA);
-		return;
 	}
 #ifdef CONFIG_WDS_WPA
 	if (hostapd_maclist_found(hapd->conf->wds_wpa_sta,
-- 
2.43.0

